docs/stack/use-cases/_index.md

```diff
@@ -0,0 +1,417 @@
+
+- If yes, the user has not signed up for campaign. Show the ad.
+- If the user clicks ad and signs up, remove the user id from that Cuckoo filter. 
+- If no, the user has signed up for that campaign. Try the next ad/Cuckoo filter. 
+ 
+**Discount code/coupon validation (retail, online shops)** 
+
+This application answers this question: Has this discount code/coupon been used yet?
+
+Use a Cuckoo filter populated with all discount codes/coupons. On every try, the entered code is checked against the filter. 
+
+- If no, the coupon is not valid. 
+- If yes, the coupon can be valid. Check the main database. If valid, remove from Cuckoo filter as `used`.
+
+### Count-Min sketch 
+
+**Products (retail, online shops)** 
+
+This application answers this question: What was the sales volume (on a certain day) for a product? 
+
+Use one Count-Min sketch created per day (period). Every product sale goes into the CMS. The CMS give reasonably accurate results for the products that contribute the most toward the sales. Products with low percentage of the total sales are ignored. 
+ 
+### Top K 
+
+**Leader boards (gaming)** 
+
+This application answers this question: Who are the K players with the highest score?
+
+Data flow is the incoming game scores. Flow id is the user id, and the value is the score. A separate sorted set is kept, storing the top K users' ids and their scores. 
+Every time a user scores points, they're added to the Top K list. 
+
+- If the result is nil, the user is already in the Top K. Update the sorted set with the user's new score. 
+- if the result is an id of another player, the player you just added (`id1`) took over the player who got returned (`id2`). Remove player `id2` from the sorted set and add player `id1`. 
+
+**Trending hashtags (social media platforms, news distribution networks)** 
+
+This application answers these questions: 
+
+- What are the K hashtags people have mentioned the most in the last X hours? 
+- What are the K news with highest read/view count today? 
+
+Data flow is the incoming social media posts from which you parse out the different hashtags. 
+
+The `TOPK.LIST` command has a time complexity of `O(K)` so if `K` is small, there is no need to keep a separate set or sorted set of all the hashtags. You can query directly from the Top K itself. 
+ 
+### HyperLogLogs 
+ 
+**Anonymous unique visits of a web page (SaaS, analytics tools)** 
+
+This application answers these questions: 
+
+- How many unique visits has this page had on this day? 
+- How many unique users have played this song? 
+- How many unique users have viewed this video? 
+
+<note><b>Note</b> Storing the IP address or any other kind of personal identifier is against the law in some countries, which makes it impossible to get unique visitor statistics on your website.</note> 
+
+One HyperLogLog is created per page (video/song) per period, and every IP/identifier is added to it on every visit.
+ 
```

Comment:
We need to add t-digest as well.

Reply:
Sounds good. I'll add that to the list of things we could add.

Reply:
Added!